### PR TITLE
added security: login, logout, bearer token and jwt

### PR DESCRIPTION
**Pull request by Borisina**

### CODE REVIEW

File: src/main/java/com/kolya/gym/config/BearerTokenFilter.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+        HttpServletResponse httpResponse = (HttpServletResponse) response;
+        String authorizationHeader = httpServletRequest.getHeader("Authorization");
+
+        if (authorizationHeader == null || !authorizationHeader.startsWith("Bearer ")) {
+            sendError(httpResponse, HttpServletResponse.SC_FORBIDDEN, "Bearer token is missing");
+            return;
+        }
+
+        String token = authorizationHeader.substring(7);
```
Comment (FloatingPirate) [inline src/main/java/com/kolya/gym/config/BearerTokenFilter.java]:
```suggestion
        String token = authorizationHeader.substring(TOKEN_IDENTIFICATION_BOUNDARY);
```
Or any other appropriate naming. 

Better not to use magic numbers. They tend to make code less readable.

File: src/main/java/com/kolya/gym/config/JwtFilter.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            throws ServletException, IOException {
+        Cookie[] cookies = request.getCookies();
+        String jwt = null;
+        String userName = null;
+        if (cookies != null) {
+            for (Cookie cookie : cookies) {
+                if (cookie.getName().equals("jwt")) {
+                    jwt = cookie.getValue();
+                    try{
+                        userName = jwtService.extractUsername(jwt);
```
Comment (FloatingPirate) [inline src/main/java/com/kolya/gym/config/JwtFilter.java]: Let's create separate method for parts of the logic. 

Too much cognitive cluttering with nested if -> for - > if.

**Review by FloatingPirate — CHANGES_REQUESTED** [review]
Good overall :) 

After fixing the suggested review comment improvements can be approved.

File: src/main/java/com/kolya/gym/config/BearerTokenFilter.java (hunk 1 above)
Comment (Borisina) [reply src/main/java/com/kolya/gym/config/BearerTokenFilter.java]: Fixed

File: src/main/java/com/kolya/gym/config/JwtFilter.java (hunk 2 above)
Comment (Borisina) [reply src/main/java/com/kolya/gym/config/JwtFilter.java]: It's better now.

**Review by FloatingPirate — APPROVED** [review]

**Pull request merged**